### PR TITLE
Add a delegate method when navigation bar poistion changed

### PR DESCRIPTION
**Pull request by OEA**
It helps developers to find out navigation bar position changes

### CODE REVIEW

File: Source/ScrollingNavigationController.swift [hunk 1]
```diff
@@ -17,6 +17,15 @@ import UIKit
...
   @objc optional func scrollingNavigationController(_ controller: ScrollingNavigationController, willChangeState state: NavigationBarState)
+  /// Called when the navigation bar position changed
+  ///
+  /// - Parameters:
+  ///   - controller: the ScrollingNavigationController
+  ///   - offset: changes amount
+  ///   - state: the new state
+  @objc optional func scrollingNavigationController(_ controller: ScrollingNavigationController,
+                                                    didUpdateOffset offset: CGFloat,
+                                                    forChangeState state: NavigationBarState)
```
Comment (andreamazz) [inline Source/ScrollingNavigationController.swift]: `forStateChange:`would be better

**Comment by andreamazz** [comment]
Hey @OEA 
Thanks for the contribution. Disregard the line length violation, apparently, I can't disable that option 🤔 
Can you change the method signature as suggested in the comment I made? Thanks

**Comment by OEA** [comment]
Yeah sure.

**Comment by andreamazz** [comment]
Appreciate it, thanks

**Comment by andreamazz** [comment]
I'll open a new release soon, I want to close another PR first. 
👍

**Pull request merged**